doc/win-dev.ps1

```diff
@@ -0,0 +1,52 @@
+Set-PSDebug -Trace 1
+
+
+$VsVersion = 2019
+$BoostVersion = @(1, 71, 0)
+$OpensslVersion = '1_1_1h'
+
+
+function Install-Exe {
+	param (
+		[string]$Url,
+		[string]$Dir
+	)
+
+	$TempDir = Join-Path ([System.IO.Path]::GetTempPath()) ([System.Guid]::NewGuid().Guid)
+	$ExeFile = Join-Path $TempDir inst.exe
+
+	New-Item -ItemType Directory -Path $TempDir
+	Invoke-WebRequest -Uri $Url -OutFile $ExeFile -UseBasicParsing
+	Start-Process -Wait -FilePath $ExeFile -ArgumentList @('/VERYSILENT', '/INSTALL', '/PASSIVE', '/NORESTART', "/DIR=${Dir}")
+	Remove-Item -Recurse -Path $TempDir
+}
+
+
+Invoke-Expression (New-Object Net.WebClient).DownloadString('https://chocolatey.org/install.ps1')
+
+$RegEnv = 'Registry::HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Session Manager\Environment'
+$ChocoPath = ";$(Join-Path $Env:AllUsersProfile chocolatey\bin)"
+
+Set-ItemProperty -Path $RegEnv -Name Path -Value ((Get-ItemProperty -Path $RegEnv -Name Path).Path + $ChocoPath)
+$Env:Path += $ChocoPath
+
+
+choco install -y "visualstudio${VsVersion}community"
+choco install -y "visualstudio${VsVersion}-workload-netcoretools"
+choco install -y "visualstudio${VsVersion}-workload-vctools"
+choco install -y "visualstudio${VsVersion}-workload-manageddesktop"
+choco install -y "visualstudio${VsVersion}-workload-nativedesktop"
+choco install -y "visualstudio${VsVersion}-workload-universal"
+choco install -y "visualstudio${VsVersion}buildtools"
+
+
+choco install -y git
+choco install -y cmake
+choco install -y winflexbison3
+choco install -y windows-sdk-8.1
+choco install -y wixtoolset
+
+
+Install-Exe -Url "https://master.dl.sourceforge.net/project/boost/boost-binaries/$($BoostVersion -join '.')/boost_$($BoostVersion -join '_')-msvc-14.2-64.exe" -Dir "C:\local\boost_$($BoostVersion -join '_')-Win64"
```

Comment:
1. Done.
2. Are you sure it works?

Reply:
Just ran `Invoke-WebRequest -Uri "https://sourceforge.net/projects/boost/files/boost-binaries/1.71.0/boost_1_71_0-msvc-14.2-64.exe/download" -OutFile "boost.exe" -UseBasicParsing -UserAgent "wget"` in PowerShell and it worked. Without setting the User-Agent, it will download the HTML page instead.

Reply:
```
PS /private/var/folders/c2/k37yy5v51qzd5_rfrwd_sd7r0000gn/T/tmp.pSsAIxwZ> Invoke-WebRequest -Uri "https://sourceforge.net/project/boost/files/boost-binaries/$($BoostVersion -join '.')/boost_$($BoostVersion -join '_')-msvc-${MsvcVersion}-64.exe/download" -OutFile boost.exe -UseBasicParsing -UserAgent wget
Invoke-WebRequest:






    var __gdpr = true;
    var __ccpa = false;
    window.bizx = window.bizx || {
        cmp: {
...
```

Reply:
I'll try to re-try the mirror failure.

![Bildschirmfoto 2020-10-29 um 15 58 03](https://user-images.githubusercontent.com/6053266/97591379-d83a8200-19ff-11eb-9585-a469142694cd.png)

Reply:
Strange, what Version of PowerShell is this? I tried with 5.1.19041.546 on Windows 10 20H2 (and in an interactive session instead of a script, but hopefully this doesn't make a difference).

Reply:
```
PS C:\Users\Administrator> Get-Host


Name             : ConsoleHost
Version          : 5.1.17763.1
InstanceId       : fc5b94f0-ed78-47f2-a24b-b6b2e924438a
UI               : System.Management.Automation.Internal.Host.InternalHostUserInterface
CurrentCulture   : en-US
CurrentUICulture : de-DE
PrivateData      : Microsoft.PowerShell.ConsoleHost+ConsoleColorProxy
DebuggerEnabled  : True
IsRunspacePushed : False
Runspace         : System.Management.Automation.Runspaces.LocalRunspace



PS C:\Users\Administrator>
```

Reply:
Seems to work. 😄

![Bildschirmfoto 2020-10-29 um 17 41 13](https://user-images.githubusercontent.com/6053266/97604899-4e45e580-1a0e-11eb-95fb-ce971846a956.png)

Reply:
OK.. too much is too much..

![Bildschirmfoto 2020-10-29 um 17 53 46](https://user-images.githubusercontent.com/6053266/97689206-8810fd80-1a9b-11eb-8b0d-92834bdd7b34.png)

Theoretically we _could_ build Boost from source, but there's also OpenSSL which _could_ vanish from upstream.

@bobapple Please mirror:

* https://master.dl.sourceforge.net/project/boost/boost-binaries/1.71.0/boost_1_71_0-msvc-14.2-64.exe
* https://slproweb.com/download/Win64OpenSSL-1_1_1h.exe

... and at best also the s/64/32/ ones and share the URLs.

Reply:
I uploaded the binaries manually, they're available here now:

* http://packages.icinga.com/windows/dependencies/boost_1_71_0-msvc-14.2-32.exe
* http://packages.icinga.com/windows/dependencies/boost_1_71_0-msvc-14.2-64.exe
* http://packages.icinga.com/windows/dependencies/Win32OpenSSL-1_1_1h.exe
* http://packages.icinga.com/windows/dependencies/Win64OpenSSL-1_1_1h.exe

It we have to update them frequently we should find a way to sync the files automatically.

Reply:
1. Thanks.
2. No, only _if we decide_ to update them.